core/cppgc.rs

```diff
@@ -125,10 +365,44 @@ pub fn try_unwrap_cppgc_object<'sc, T: GarbageCollected + 'static>(
   let obj =
     unsafe { v8::Object::unwrap::<CPPGC_TAG, CppGcObject<T>>(isolate, obj) }?;
 
-  if obj.tag != TypeId::of::<T>() {
+  Some(Ptr {
+    inner: obj,
+    root: None,
+  })
+}
+
+#[doc(hidden)]
+#[allow(clippy::needless_lifetimes)]
+pub fn try_unwrap_cppgc_proto_object<
+  'sc,
+  T: GarbageCollected + PrototypeChain + 'static,
+>(
+  isolate: &mut v8::Isolate,
+  val: v8::Local<'sc, v8::Value>,
+) -> Option<Ptr<T>> {
+  let Ok(obj): Result<v8::Local<v8::Object>, _> = val.try_into() else {
+    return None;
+  };
+  if !obj.is_api_wrapper() {
     return None;
   }
 
+  let obj = if let Some(proto_index) = T::prototype_index() {
+    let proto_chain = unsafe {
+      v8::Object::unwrap::<CPPGC_TAG, PrototypeChainStore>(isolate, obj)
```

Comment:
need to use a different tag for prototype objects, otherwise you get type confusion between them and normal cppgc objects.